### PR TITLE
Pretty printer - remove sequential blank lines

### PR DESCRIPTION
**Pull request by FreHu**
Removes any blank lines above the limit specified in abaplint.json.

rftT7wMLRW

- also include lines containing only whitespace (for sequential_blanks as well)
- moved pretty printing operations to separate classes

### CODE REVIEW

File: src/rules/whitespace/sequential_blank.ts [hunk 1]
```diff
@@ -39,7 +42,7 @@ export class SequentialBlank extends ABAPRule {
     let blanks = 0;
 
     for (let i = 0; i < rows.length; i++) {
-      if (rows[i] === "") {
+      if (SequentialBlank.isBlankOrWhitespace(rows[i])) {
```
Comment (g-back) [inline src/rules/whitespace/sequential_blank.ts]: super minor thing, but this if-block could be shortened with:
```typescript
blanks = SequentialBlank.isBlankOrWhitespace(rows[i]) ? blanks++ : blanks = 0;
```

Comment (FreHu) [reply src/rules/whitespace/sequential_blank.ts]: I don't like having side effects in ternaries, the line does a bit too much. But I did improve the code a bit differently.

**Comment by larshp** [comment]
I would like to keep the pretty printer under /abap/, but I guess it can be moved sometime in the future.

thinking of having multiple pretty printers, one for ABAP + CDS + XSLT + more, but it might also be okay to put it all under pretty_printer 🤷‍♂

**Comment by larshp** [comment]
thanks 👍

**Pull request merged**